File: Data Structures & Algorithms/palindromic-substrings/submission-4.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    countSubstrings(s) {
        const n = s.length;

        let ans = 0;
        const expand = (l, r) => {
            while(l >= 0 && r < n && s[l] == s[r]) {
                ans++;
                l--, r++;
            }
        }

        for(let i = 0; i < n; ++i) {
            expand(i, i);
            expand(i, i + 1);
        }

        return ans;
    }
}

File: Data Structures & Algorithms/merge-triplets-to-form-target/submission-4.py
class Solution:
    def mergeTriplets(self, triplets: List[List[int]], target: List[int]) -> bool:
        ans = 0
        for a, b, c in triplets:
            if a <= target[0] and b <= target[1] and c <= target[2]:
                if a == target[0]:
                    ans |= 1
                if b == target[1]:
                    ans |= 2
                if c == target[2]:
                    ans |= 4
        return ans == 7